src/main/java/carpet/script/CarpetExpression.java

```diff
@@ -2193,52 +2198,60 @@ private void API_IteratingOverAreasOfBlocks()
             int smaxx;
             int smaxy;
             int smaxz;
-            try
-            {
-                cx = (int)((NumericValue) lv.get(0).evalValue(c)).getLong();
-                cy = (int)((NumericValue) lv.get(1).evalValue(c)).getLong();
-                cz = (int)((NumericValue) lv.get(2).evalValue(c)).getLong();
-                if (lv.size()==3) // only done this way because of stupid Java lambda final reqs
-                {
-                    sminx = 1;
-                    sminy = 1;
-                    sminz = 1;
-                    smaxx = 1;
-                    smaxy = 1;
-                    smaxz = 1;
-                }
-                else if (lv.size()==6)
-                {
-                    sminx = (int) ((NumericValue) lv.get(3).evalValue(c)).getLong();
-                    sminy = (int) ((NumericValue) lv.get(4).evalValue(c)).getLong();
-                    sminz = (int) ((NumericValue) lv.get(5).evalValue(c)).getLong();
+
+            BlockArgument cposLocator = BlockArgument.findIn(cc, lv, 0);
+            BlockPos cpos = cposLocator.block.getPos();
+
+            cx = cpos.getX();
+            cy = cpos.getY();
+            cz = cpos.getZ();
+
+            try{
+                BlockArgument diff_Locator = BlockArgument.findIn(cc, lv, cposLocator.offset);
+                BlockPos difference = diff_Locator.block.getPos();
+
+                sminx = difference.getX();
+                sminy = difference.getY();
+                sminz = difference.getZ();
+
+                try{
+                    BlockPos pos_diff=BlockArgument.findIn(cc,lv,diff_Locator.offset).block.getPos();
+                    smaxx = pos_diff.getX();
+                    smaxy = pos_diff.getY();
+                    smaxz = pos_diff.getZ();
+                    Messenger.m(cc.s,"w 9");
+                } catch(IndexOutOfBoundsException i){//Checing if there are 2 pos
                     smaxx = sminx;
                     smaxy = sminy;
                     smaxz = sminz;
                 }
-                else // size == 9
-                {
-                    sminx = (int) ((NumericValue) lv.get(3).evalValue(c)).getLong();
-                    sminy = (int) ((NumericValue) lv.get(4).evalValue(c)).getLong();
-                    sminz = (int) ((NumericValue) lv.get(5).evalValue(c)).getLong();
-                    smaxx = (int)((NumericValue) lv.get(6).evalValue(c)).getLong();
-                    smaxy = (int)((NumericValue) lv.get(7).evalValue(c)).getLong();
-                    smaxz = (int)((NumericValue) lv.get(8).evalValue(c)).getLong();
-                }
-            }
-            catch (ClassCastException exc)
-            {
-                throw new InternalExpressionException("Attempted to pass a non-number to rect");
+
+            } catch (IndexOutOfBoundsException i){//Checking if there is only 1 pos
```

Comment:
using try/catch for regular logic is very detrimental to the performance. Use try-catches only to catch errors that would flip the script anyways.

Use if (lv.size() > locator.offset).... to continue with optional parameters